Feature/first iteration

anuragkainth opened the pull request:
- Added .devcontainer with all the necessary configurations for Vercel AI SDK, Svelte using typescript.
- Added readme with all required details regarding the configuration and how to create Daytona workspace for this repo.
- Added .env.example to help users configure Groq API in the application.
- Configured llama-3.1 model for AI responses using Groq.

anuragkainth commented:
- Updated the UI for the Chatbot as per Daytona's theme.
- Updated [readme](https://github.com/daytonaio-experiments/starter-ai-chatbot/blob/main/README.md ) in main branch with new Chat UI.

nibzard commented on src/routes/api/assistant/+server.ts:
```diff
@@ -0,0 +1,110 @@
+  apiKey: env.GROQ_API_KEY || '',
+});
+
+const homeTemperatures = {
```
what is this, please cleanup the code

anuragkainth replied:
sure, i'll make sure to clean this in next commit.

nibzard commented on src/routes/+page.svelte:
```diff
@@ -0,0 +1,373 @@
+
+<div class="page-container">
+  <aside class="sidebar">
+    <img src="/logo.svg" alt="Logo" class="logo" />
```
please remove the daytona logo

anuragkainth replied:
noted. will remove this in next commit.

nibzard commented on src/routes/+page.svelte:
```diff
@@ -0,0 +1,373 @@
+    </form>
+  </section>
+</div>
+
```
please add footer with CTA: Simplify your dev environment with Daytona (and link github repo)

anuragkainth replied:
sure, will make these changes.

nibzard commented on src/routes/api/chat/+server.ts:
```diff
@@ -0,0 +1,20 @@
+
+  const result = await streamText({
+    model: groq('llama-3.1-8b-instant'),
+    messages: convertToCoreMessages(messages),
```
please add system prompt and explain it in the article

for example something like:
  system:
    `You are helpful assistant from Daytona with senior level expertise in development. ` +
    `Respond to the users' request with a clear and concise answer ` +
    `about any of his development or devops problems.`,

anuragkainth replied:
sure, i will add system prompt and make sure of these changes.

nibzard commented on src/routes/api/use-chat-tools/+server.ts:
```diff
@@ -0,0 +1,44 @@
+    model: groq('llama-3.1-8b-instant'),
+    messages: convertToCoreMessages(messages),
+    tools: {
+      getWeatherInformation: {
```
?

anuragkainth replied:
All these were present in the reference repo. We have just updated the Chat bot section for now. I will clean all these in next commit.

nibzard commented on src/routes/chat-with-tools/+page.svelte:
```diff
@@ -0,0 +1,57 @@
+    maxToolRoundtrips: 5,
+    // run client-side tools that are automatically executed:
+
+    async onToolCall({ toolCall }) {
```
?

anuragkainth replied:
we'll remove all unnecessary parts in next commit.

nibzard commented on src/routes/chat-with-tools/+page.svelte:
```diff
@@ -0,0 +1,57 @@
+
+                  {#if 'result' in toolInvocation}
+                      <div>
+                          Tool call {`${toolInvocation.toolName}: `}
```
tool calls are very interesting, will you use them in article? would be nice, otherwise remove

anuragkainth replied:
this most probably will be removed in next iteration from the repo.

nibzard commented on src/routes/completion/+page.svelte:
```diff
@@ -0,0 +1,37 @@
+
+<svelte:head>
+	<title>Home</title>
+	<meta name="description" content="Svelte demo app" />
```
be consistent and cleanup the code

anuragkainth replied:
noted. will do a code cleanup and other changes we discussed in next iteration. Thankyou for this review.

nkkko reviewed (changes requested):
please cleanup the code

anuragkainth commented:
- code cleanup and refactoring.
- removed daytona logo and replaced "Voltona" with "AI Chat Demo".
- removed new chat, attachment and voice buttons.
- added footer with CTA as per comments.
- added system prompt as per the comments in `/src/routes/api/chat/+server.ts`.
- updated readme in [main branch](https://github.com/daytonaio-experiments/starter-ai-chatbot/blob/main/README.md) with updated image and compressed `--code` command.
